### PR TITLE
[argumentum] Add new port

### PR DESCRIPTION
**Pull request by mmahnic**
Argumentum is a library for parsing command line arguments. It can be used as a static library and as a header-only library. A dynamic version is planned for a future release.

    What does your PR fix? It adds a new port.

    Which triplets are supported/not supported? Have you updated the CI baseline?
    Tested only on x64-linux.

    Does your PR follow the maintainer guide?
    I believe so.

### CODE REVIEW

File: ports/argumentum/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mmahnic/argumentum
+    REF v0.2.1
+    SHA512 e2d452b5e4aa37a5b76b14ac905e040e1f0791aa687f912addcf949c64747aee21273a345a99851f64c9fad337cd11f74c408a3c6455346c30eea1aaee2244a6
+    HEAD_REF master
+)
+
+include( GNUInstallDirs )
```
Comment (NancyLi1013) [inline ports/argumentum/portfile.cmake]: `include( GNUInstallDirs )` should be not needed.
Could you please remove this?

File: ports/argumentum/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+    -DARGUMENTUM_BUILD_EXAMPLES=OFF
+    -DARGUMENTUM_BUILD_TESTS=OFF
+)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH ${CMAKE_INSTALL_LIBDIR}/cmake/Argumentum)
```
Comment (NancyLi1013) [inline ports/argumentum/portfile.cmake]: This line may be updated as
`vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/Argumentum)`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @mmahnic thanks for this PR.
I noticed that this port failed on `arm-uwp, x64-uwp, arm64-windows`.
Are those expected? If this port currently doesn't support `arm, arm64, uwp`, please add this to portfile.cmake.

`vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "uwp")
`
Besides this, please also update [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) with the following content:
```
argumentum:arm4-windows=fail
argumentum:x64-uwp=fail
argumentum:arm-uwp=fail
```
If not, please try to fix the failures.

File: ports/argumentum/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,29 @@
+include(vcpkg_common_functions)
+
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mmahnic/argumentum
```
[the remark this answers is not in the document]
Comment (mmahnic) [reply ports/argumentum/portfile.cmake]: This is the original repository.

File: ports/argumentum/portfile.cmake (hunk 2 above)
Comment (mmahnic) [reply ports/argumentum/portfile.cmake]: I have applied all the suggested changes.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**